### PR TITLE
Add WarpX

### PR DESCRIPTION
**Pull request by ax3l**
Add a first build for WarpX.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [ ] Package does not vendor other packages: [AMReX](https://github.com/AMReX-Codes/amrex) & [PICSAR](https://github.com/ECP-WarpX/picsar) are pulled on the fly, under the same license. Many binary variants make separating them very hard for now.
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [ ] When in trouble be please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by RemiLehe** [comment]
I approve to be co-maintainer of this feedstock.

**Comment by ax3l** [comment]
Trying to address:
```
  ERROR (warpx,bin/warpx.3d.MPI.OMP.DP.OPMD.QED): $RPATH/libopenPMD.so not found in packages, sysroot(s) nor the missing_dso_whitelist.
  ERROR (warpx,bin/warpx.3d.MPI.OMP.DP.OPMD.QED): $RPATH/libmpi.so.40 not found in packages, sysroot(s) nor the missing_dso_whitelist.
```

**Comment by ax3l** [comment]
Hi @conda-forge/help-c-cpp, do you know what could be wrong here in the Linux packaging step? I tried my best, but stay confused :)

File: recipes/warpx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make          # [unix]
+    - cmake >=3.14.0
+    - libgomp       # [linux]
+    - llvm-openmp   # [osx]
+  host:
+    - boost-cpp >=1.66.0
```
Comment (isuruf) [inline recipes/warpx/meta.yaml:30]: To use global conda-forge pinning
```suggestion
    - boost-cpp
```

**Comment by ax3l** [comment]
@isuruf uff, still has a problem to catch a dependency

**Comment by isuruf** [comment]
That's because https://github.com/conda-forge/openpmd-api-feedstock/blob/ca07539ab136039d1dcd6a4c9954ed612b38b5b2/recipe/meta.yaml#L52 is causing issues. Maybe changing it to `min_pin='x.x.x.x'` will work

**Comment by ax3l** [comment]
Thanks for the hint - does it harm to put the 4th digit on `max_pin`, too? https://github.com/conda-forge/openpmd-api-feedstock/pull/57

**Comment by ax3l** [comment]
@isuruf it passes, thank you for your help!

Will enable Windows later on, needs a few upstream fixes that I'll land in the next release instead of patching now.

**Pull request merged**